### PR TITLE
feat: add tweet links

### PR DESCRIPTION
**Pull request by fersilva16**
# Summary

image

### CODE REVIEW

File: apps/web/src/components/tweetContent.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  type: 'text' | 'mention' | 'url';
+  index: number;
+  value: string;
+};
+
+const matchAll = <T extends string>(text: string, type: T, regex: RegExp) => {
+  return [...text.matchAll(regex)].map((match) => ({ type, match }));
+};
+
+export const parseContent = (text: string) => {
```
Comment (sibelius) [inline apps/web/src/components/tweetContent.ts:14]: is there a better way?

File: apps/web/src/components/TweetInfo.tsx [hunk 2]
```diff
@@ -65,13 +79,58 @@ const TweetInfo = ({ tweet }: TweetInfoProps) => {
...
-          <Text mt={2}>{decodeHTML(tweet.text)}</Text>
-          <Text color="gray.500" marginTop={2}>
-            {tweetTimeInfo()}
-          </Text>
+          </Link>
         </Flex>
-      </Link>
+        <Box zIndex={1} pointerEvents="none">
+          <Text mt={2}>
+            {parseContent(decodeHTML(tweet.text)).map((chunk) => {
```
Comment (sibelius) [inline apps/web/src/components/TweetInfo.tsx:86]: should we always parse on every render?

what about moving this to another component?

File: apps/web/src/components/TweetInfo.tsx [hunk 3]
```diff
@@ -65,13 +79,58 @@ const TweetInfo = ({ tweet }: TweetInfoProps) => {
...
+              if (chunk.type === 'text') return chunk.value;
+
+              if (chunk.type === 'url') {
```
Comment (sibelius) [inline apps/web/src/components/TweetInfo.tsx:89]: what about 1 component per chunk type?

<TweetChunkUrl />

Comment (sibelius) [reply apps/web/src/components/TweetInfo.tsx:89]: try to follow this pattern https://medium.com/reactbrasil/formik-como-criar-um-formul%C3%A1rio-com-campos-din%C3%A2micos-a150d60a8610

File: apps/web/tsconfig.json [hunk 4]
```diff
@@ -15,6 +15,7 @@
     "jsx": "preserve",
     "incremental": true,
     "baseUrl": "src",
+    "downlevelIteration": true
```
Comment (sibelius) [inline apps/web/tsconfig.json:18]: what is this about?

**Review by sibelius — APPROVED** [review]
should we add 1 text per chunk type?

File: apps/web/src/components/TweetInfo.tsx (hunk 2 above)
Comment (danielcti) [reply apps/web/src/components/TweetInfo.tsx:86]: maybe the useMemo can fix the unnecessary rerenders

File: apps/web/src/components/tweetContent.ts (hunk 1 above)
Comment (fersilva16) [reply apps/web/src/components/tweetContent.ts:14]: I think a lexer is better, I didn't implement that because maybe it's over-engineering.

**Pull request merged**